### DISCUSSION TITLE
Lazer reprocessing all maps just to add 10 missing beatmaps

### DISCUSSION THREAD

**Discussion by Mrcubix** [ppy/osu#27045] [Ideas]
### Type

Other

### Bug description

Lazer currently allow importing from stable in order synchronize the user data of both clients, but what if you downloaded an osz, osk or osr file, imported it into stable (since it's associated to stable by default), but also want it to get imported into lazer?

You could make a copy of the osk in advance, but you would need to develop the muscle memory for that.
You could go in stable, go to the beatmap page for that map and then download it again and drag it into lazer, but it's annoying to have to do that every time.
And finally, the solution mentioned by ppy on multiple occasions: you could also re-import user data, but you'll be forced to sit through lazer reprocessing thousands of maps, just to find out all it had to do was import a dozen maps.

I think the later is a big issue, I shouldn't have to wait an hour to synchronize my maps, especially when, again, only a dozen were added.

### Screenshots or videos

![re-processing](https://i.imgur.com/aw00hQh.png)
![newly added maps in stable](https://i.imgur.com/dgIVKTq.png)

### Version

2024.131.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14157997/compressed-logs.zip)

**Comment by bdach** [comment]
The game is not importing every single beatmap. Due to stable's file structure a few bare minimum checks must be however performed to make sure that no beatmap has been messed with in ways that do not preserve their integrity.

> but what if you downloaded an osz, osk or osr file, imported it into stable (since it's associated to stable by default)

File association support will hopefully be coming soon.

I'm gonna move this to a discussion thread because this is really a feature request for a sync facility.

**Reply by rosacry** [reply]
Would a narrower version of this be worth pursuing?

My use case is that I imported my stable beatmaps into lazer once, then kept adding more maps to stable afterward. Re-running the stable beatmap import works, but with a large stable library it still has to check a lot of already-imported content just to pick up the newer maps.

I am wondering whether an incremental stable beatmap import improvement would be acceptable to work on, rather than trying to build a full stable/lazer sync facility. The goal would be to make later beatmap imports faster while keeping whatever integrity checks are still necessary.

If that direction is not wanted, is there a preferred design or existing issue/discussion I should follow before spending time investigating it?

**Reply by cihe13375** [reply]
I think you can try to write a small program that automatically make a copy the .osz archive, and call stable and lazer on the two archives respectively. Then set the file association of .osz to this small program. This needs some coding, but I think it won't be too difficult.
